### ISSUE TITLE
sum+distinct not support

### ISSUE THREAD

**Issue by dbzzzzz** [apache/shardingsphere#10564]
### Which version of ShardingSphere did you use?
ShardingSphere  5
### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy
### Expected behavior
select sum(distinct user_id),sum(order_id) from t_order; 
should return correct result.
### Actual behavior
mysql> select sum(distinct user_id),sum(order_id) from t_order;
ERROR 1140 (42000): In aggregated query without GROUP BY, expression #1 of SELECT list contains nonaggregated column 'testdb_0.t_order_0.user_id'; this is incompatible with sql_mode=only_full_group_by

**Comment by strongduanmu** [comment]
This exception is caused by SS rewriting. SS will rewrite `sum(distinct user_id)` into `distinct user_id` to execute, the routing result is as follows, if `distinct user_id` and `sum(order_id)` are executed at the same time, an error will occur.

actual sql:

```sql
[INFO ] 2021-05-31 18:22:54.567 [Connection-3-ThreadExecutor] ShardingSphere-SQL - Logic SQL: select sum(distinct user_id),sum(order_id) from t_order;
[INFO ] 2021-05-31 18:22:54.567 [Connection-3-ThreadExecutor] ShardingSphere-SQL - SQLStatement: PostgreSQLSelectStatement(limit=Optional.empty, lock=Optional.empty)
[INFO ] 2021-05-31 18:22:54.567 [Connection-3-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_0 ::: select DISTINCT user_id AS AGGREGATION_DISTINCT_DERIVED_0,sum(order_id) from t_order_0;
[INFO ] 2021-05-31 18:22:54.567 [Connection-3-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_0 ::: select DISTINCT user_id AS AGGREGATION_DISTINCT_DERIVED_0,sum(order_id) from t_order_1;
[INFO ] 2021-05-31 18:22:54.567 [Connection-3-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_1 ::: select DISTINCT user_id AS AGGREGATION_DISTINCT_DERIVED_0,sum(order_id) from t_order_0;
[INFO ] 2021-05-31 18:22:54.567 [Connection-3-ThreadExecutor] ShardingSphere-SQL - Actual SQL: ds_1 ::: select DISTINCT user_id AS AGGREGATION_DISTINCT_DERIVED_0,sum(order_id) from t_order_1;
```

**Comment by strongduanmu** [comment]
Fixed by #10856.
